packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -75,11 +80,12 @@ export class DataStores implements IDisposable {
     ) {
         this.logger = ChildLogger.create(baseLogger);
         // Extract stores stored inside the snapshot
-        const fluidDataStores = new Map<string, ISnapshotTree | string>();
+        const fluidDataStores = new Map<string, ISnapshotTree>();
 
-        if (typeof baseSnapshot === "object") {
+        const nonDataStorePathsToUse = baseSnapshotType === "legacy" ? nonDataStorePaths : [];
```

Comment:
i'd like all the backcompat logic in one place. if we put the previous logic into a factory function, you could also create a tree there that strips nonDataStorePaths, which makes this code simpler

Reply:
the we shouldn't need BaseSnapshotType at all

Reply:
Removed BaseSnapshotType by stripping out non-datastore path parts from a shallow-copy of the tree. Thanks for pointing this out.